EIPS/eip-7775.md

```diff
@@ -12,62 +12,38 @@ created: 2024-09-30
 2. Retrieves the current address from the EVM execution context.
 3. Checks the balance of the current address.
 4. If the amount to be burned is greater than the balance of the current address, the opcode `MUST` revert.
 5. If the amount to be burned is 0, the execution `MUST NOT` revert.
-6. Subtracts the amount from the current address's native token balance.
+6. Subtracts the amount from the current address's native ether balance.
 
 When `BURN` is used in the context of a DELEGATECALL or CALLCODE, the contract whose balance is to be manipulated is the contract that issued the DELEGATECALL or CALLCODE instruction.
 
 When `BURN` is used in the context of a STATICCALL the call `MUST` revert.
 
-Example pseudocode for the `BURN` opcode:
-
-```python
-def op_burn(pc, interpreter, scope):
-    # Pop the value to be burned from the stack
-    value_to_burn = scope.stack.pop()
-    
-    # Retrieve the current address from the EVM execution context
-    current_address = scope.contract.address()
-    
-    # Check the balance of the current address
-    balance = interpreter.evm.state_db.get_balance(current_address)
-    
-    # If the value to be burned is greater than the balance, revert
-    if value_to_burn > balance:
-        return "ErrInsufficientBalance"
-    
-    # If the value to be burned is 0, do not revert
-    if value_to_burn == 0:
-        return None
-    
-    # Subtract the value from the current address's balance
-    interpreter.evm.state_db.sub_balance(current_address, value_to_burn)
-    return None
-```
-
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+When the `BURN` opcode is processed, 0 gas MUST be consumed.
 
 ## Rationale
 
-The introduction of the `BURN` opcode helps clean up a piece of weird semantics in the Ethereum. Historically, burning native tokens involved sending them to an address from which they could not be recovered, such as the zero address or a contract with no withdrawal functionality. This method is not only inefficient but also confusing for indexers and other tools that track token movements. By providing a dedicated `BURN` opcode, we eliminate this ambiguity and ensure that the act of burning tokens is explicit and standardized.
+The introduction of the `BURN` opcode helps clean up a piece of weird semantics in the Ethereum. Historically, burning native ether involved sending them to an address from which they could not be recovered, such as the zero address or a contract with no withdrawal functionality. This method is not only inefficient but also confusing for indexers and other tools that track token movements. By providing a dedicated `BURN` opcode, we eliminate this ambiguity and ensure that the act of burning tokens is explicit and standardized.
```

Comment:
(Informational): This is not entirely correct, in the past it was actually possible to burn ether. In order to do so, you had to SELFDESTRUCT to the current address. In https://eips.ethereum.org/EIPS/eip-6780 this was removed, see:

> Note that if the target is the same as the contract calling SELFDESTRUCT there is no net change in balances. Unlike the prior specification, Ether will not be burnt in this case.

(This info should not be added, but just wanted to note :) )

Reply:
Very interesting!